### PR TITLE
[ScrollView] Don't always adjust content view padding for both scrollers

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

Based on your system preferences, NSScrollView may or may not always show it's scrollbars. When the scrollbars are shown, the _content view_ of the scroll view is smaller than the scroll view itself, depending on whether we show a horizontal scrollbar, a vertical one, or both. Yoga doesn't know about this shrinking of content view's size, so we get around this by adding padding to the content view if scroll bars are visible.

The issue was, we would add padding for _both_ horizontal and vertical scrollbars, event if only one or the other was shown. We fix that here with this change. Previously we passed the entire NSScroller (both vertical and horizontal) to `RCTScollContentLocalData` so that we could read the height/width. According to [Apple documentation](https://developer.apple.com/documentation/appkit/nsscrollview/1403447-horizontalscroller), an NSScroller may be allocated even if it's not shown. So, instead of passing the whole scroller, let's just see if it's shown and pass the width/height instead.

## Changelog

[macOS] [Fixed] - ScrollView - Don't always adjust content view padding for both scrollers

## Test Plan

Ran RN-Tester and scrollviews appeared as expected.

### CODE REVIEW

**Comment by huwilkes** [comment]
Nit: "NSScroller may be allovated" -> allocated

File: React/Views/ScrollView/RCTScrollContentView.m [hunk 1]
```diff
@@ -61,10 +61,13 @@ - (void)reactSetFrame:(CGRect)frame
       _hasHorizontalScroller = nextHasHorizontalScroller;
       _hasVerticalScroller = nextHasVerticalScroller;
 
+      CGFloat horizontalScrollerHeight = _hasHorizontalScroller ? NSWidth([[platformScrollView horizontalScroller] frame]) : 0;
```
Comment (huwilkes) [inline React/Views/ScrollView/RCTScrollContentView.m:64]: Should this be NSHeight instead of NSWidth?

**Review by huwilkes — APPROVED** [review]

File: React/Views/ScrollView/MacOS/RCTScrollContentLocalData.h [hunk 2]
```diff
@@ -14,8 +14,8 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, assign) CGFloat horizontalScrollerHeight;
 @property (nonatomic, assign) CGFloat verticalScrollerWidth;
 
-- (instancetype)initWithVerticalScroller:(nullable NSScroller *)verticalScroller
-                      horizontalScroller:(nullable NSScroller *)horizontalScroller;
+- (instancetype)initWithVerticalScrollerWidth:(CGFloat)verticalScrollerWidth
+										 horizontalScrollerHeight:(CGFloat)horizontalScrollerHeight;
```
Comment (harrieshin) [inline React/Views/ScrollView/MacOS/RCTScrollContentLocalData.h:18]: what's up with this indentation style?

Comment (Saadnajmi) [reply React/Views/ScrollView/MacOS/RCTScrollContentLocalData.h:18]: spcaes got converted to tabs... fixing

**Pull request merged**